src/main/java/makePost/use_case/MakePostInteractor.java

```diff
@@ -0,0 +1,107 @@
+package makePost.use_case;
+
+import entities.Post;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static java.time.temporal.ChronoUnit.DAYS;
+
+public class MakePostInteractor implements MakePostInputBoundary {
+    private final MakePostDataAccessInterface dataAccess;
+    private final MakePostOutputBoundary presenter;
+
+    public MakePostInteractor(MakePostDataAccessInterface dataAccess, MakePostOutputBoundary presenter){
+        this.dataAccess = dataAccess;
+        this.presenter = presenter;
+    }
+
+    /**
+     * Checks whether the deadline is at most six months from the creationDate.
```

Comment:
should we also check that the deadline hasn't passed?

Reply:
Done.